lib/framescript/selection-events.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const { classes: Cc, interfaces: Ci } = Components;
+const obsSvc = Cc["@mozilla.org/observer-service;1"].getService(Ci.nsIObserverService);
+
+const INTERACTIVE = Ci.nsISelectionListener.MOUSEUP_REASON |
+                    Ci.nsISelectionListener.KEYPRESS_REASON |
+                    Ci.nsISelectionListener.SELECTALL_REASON;
+
+let selections = new WeakMap();
+let isPrivate = docShell.QueryInterface(Ci.nsILoadContext).usePrivateBrowsing;
+
+obsSvc.addObserver(observer, 'document-shown', false);
+obsSvc.addObserver(observer, 'document-element-inserted', false);
```

Comment:
Might be worth testing if using the DOM event "DOMWindowCreated" is better than using global observer notifications.